numpy/_core/numeric.py

```diff
@@ -1163,13 +1163,13 @@ def tensordot(a, b, axes=2):
     notin = [k for k in range(nda) if k not in axes_a]
     newaxes_a = notin + axes_a
     N2 = math.prod(as_[axis] for axis in axes_a)
-    newshape_a = (math.prod([as_[ax] for ax in notin]), N2)
+    newshape_a = (math.prod(as_[ax] for ax in notin), N2)
     olda = [as_[axis] for axis in notin]
 
     notin = [k for k in range(ndb) if k not in axes_b]
     newaxes_b = axes_b + notin
     N2 = math.prod(bs[axis] for axis in axes_b)
-    newshape_b = (N2, math.prod([bs[ax] for ax in notin]))
+    newshape_b = (N2, math.prod(bs[ax] for ax in notin))
```

Comment:
These are the only changes that *may* be possibly impactful on performance, and I agree that it shouldn't matter over other stuff that's happening in this function. The other use outside of distutils and tests is in a `__repr__` implementation, and printing performance isn't a big deal either.

IMO this change is fine and is marginally more readable.

Reply:
How do you think about https://github.com/numpy/numpy/pull/27884#issuecomment-2515346047 ?